src/iuse_software_kitten.cpp

```diff
@@ -295,7 +298,10 @@ robot_finds_kitten::robot_finds_kitten(WINDOW *w)
         do {
             bogus[c].character = ktile[rand() % 82];
         } while (bogus[c].character == '#' || bogus[c].character == ' ');
-        bogus[c].color = int_to_color((rand() % (maxcolor - 2)) + 2);
+
+        do {
+            bogus[c].color = all_colors.get_random();
+        } while ( all_colors.get(kitten.color) == "c_black" );
```

Comment:
1. You meant to test `bogus[c].color`.
2. How about a function `all_colors.get_random_non_black()` or similar?